examples/bsoncxx/builder_core.cpp

```diff
@@ -0,0 +1,55 @@
+#include <bsoncxx/builder/core.hpp>
+#include <bsoncxx/types.hpp>
+
+using namespace bsoncxx;
+
+int main(int, char**) {
+
+    // bsoncxx::builder::core is a low-level primitive that can be useful for building other
+    // BSON abstractions. Most users should just use builder::stream or builder::basic.
+
+    // the boolean argument to core should be true if the top-level BSON datum is an array.
+    // if false, it will be created as a document.
+    auto builder = builder::core{false};
```

Comment:
Actually, I mean to point this out too. Thanks for raising this @TylerBrock . Should we make this an enum or something? (In another commit, open a ticket, etc.).